ee/katc/table.go

```diff
@@ -39,21 +43,60 @@ func newKatcTable(tableName string, cfg katcTableConfig, slogger *slog.Logger) (
 		columnLookup[cfg.Columns[i]] = struct{}{}
 	}
 
-	return &katcTable{
-		cfg:          cfg,
-		columnLookup: columnLookup,
-		slogger: slogger.With(
-			"table_name", tableName,
-			"table_type", cfg.SourceType,
-			"table_source", cfg.Source,
-		),
-	}, columns
+	k := katcTable{
+		sourceType:        cfg.SourceType,
+		sourcePaths:       cfg.SourcePaths,
+		sourceQuery:       cfg.SourceQuery,
+		rowTransformSteps: cfg.RowTransformSteps,
+		columnLookup:      columnLookup,
+		slogger:           slogger,
+	}
+
+	// Check overlays to see if any of the filters apply to us;
+	// use the overlay definition if so.
+	for _, overlay := range cfg.Overlays {
+		if !filtersMatch(overlay.Filters) {
+			continue
+		}
+
+		if overlay.SourceType != nil {
+			k.sourceType = *overlay.SourceType
+		}
+		if overlay.SourcePaths != nil {
+			k.sourcePaths = *overlay.SourcePaths
+		}
+		if overlay.SourceQuery != nil {
+			k.sourceQuery = *overlay.SourceQuery
+		}
+		if overlay.RowTransformSteps != nil {
+			k.rowTransformSteps = *overlay.RowTransformSteps
+		}
+
+		break
+	}
+
+	// Add extra fields to slogger
+	k.slogger = slogger.With(
+		"table_name", tableName,
+		"table_type", cfg.SourceType,
+		"table_source_paths", cfg.SourcePaths,
```

Comment:
```suggestion
		"table_type", cfg.SourceType,
```
I'd probably leave this out -- I think we're likely to path large arrays of them in. There's value in know what path actually matches on disk. but probably not the full paths array